powerstore/datasource_filesystem.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ datasource.DataSource              = &fileSystemDataSource{}
+	_ datasource.DataSourceWithConfigure = &fileSystemDataSource{}
+)
+
+// newFileSystemDataSource returns the fileSystem data source object
+func newFileSystemDataSource() datasource.DataSource {
+	return &fileSystemDataSource{}
+}
+
+// fileSystemDataSource is the data source implementation
+type fileSystemDataSource struct {
+	client *client.Client
+}
+
+// Metadata returns the data source type name
+func (d *fileSystemDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_filesystem"
+}
+
+// Schema defines the schema for the data source
+func (d *fileSystemDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description:         "This datasource is used to query the existing File System from PowerStore array. The information fetched from this datasource can be used for getting the details for further processing in resource block.",
+		MarkdownDescription: "This datasource is used to query the existing File System from PowerStore array. The information fetched from this datasource can be used for getting the details for further processing in resource block.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Unique identifier of the File System. Conflicts with `name` and `nas_server_id`.",
+				MarkdownDescription: "Unique identifier of the File System. Conflicts with `name` and `nas_server_id`.",
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("name"), path.MatchRoot("nas_server_id")),
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description:         "File System name. Conflicts with `id` and `nas_server_id`.",
+				MarkdownDescription: "File System name. Conflicts with `id` and `nas_server_id`.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.ConflictsWith(path.MatchRoot("id"), path.MatchRoot("nas_server_id")),
+				},
+			},
+			"nas_server_id": schema.StringAttribute{
+				Description:         "Nas server ID. Conflicts with `id` and `name`.",
+				MarkdownDescription: "Nas server ID. Conflicts with `id` and `name`.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.ConflictsWith(path.MatchRoot("id"), path.MatchRoot("name")),
```

Comment:
I think check is not needed since it already added to other 2 attributes

Reply:
done